sc-kpm/sc-agents-common/agents/GenerationByTemplateAgent.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022 Intelligent Semantic Systems LLC, All rights reserved.
+ */
+
+#include "sc-memory/scs/scs_types.hpp"
+
+#include "utils/CommonUtils.hpp"
+#include "utils/IteratorUtils.hpp"
+
+#include "GenerationByTemplateAgent.hpp"
+
+namespace scAgentsCommon
+{
+SC_AGENT_IMPLEMENTATION(GenerationByTemplateAgent)
+{
+  ScAddr questionNode = m_memoryCtx.GetEdgeTarget(edgeAddr);
+  if (!checkActionClass(questionNode))
+    return SC_RESULT_OK;
+
+  SC_LOG_DEBUG("GenerationByTemplateAgent started")
+
+  ScAddr templateLinkAddr = utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, CoreKeynodes::rrel_1);
+  ScAddr parameterNodeAddr =
+      utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, CoreKeynodes::rrel_2);
+
+  if (!templateLinkAddr.IsValid())
+  {
+    SC_LOG_ERROR("GenerationByTemplateAgent: template link not found.")
+    utils::AgentUtils::finishAgentWork(&m_memoryCtx, questionNode, false);
+    return SC_RESULT_ERROR;
+  }
+  if (!parameterNodeAddr.IsValid())
+  {
+    SC_LOG_ERROR("GenerationByTemplateAgent: parameter node not found.")
+    utils::AgentUtils::finishAgentWork(&m_memoryCtx, questionNode, false);
+    return SC_RESULT_ERROR;
+  }
+
+  std::string linkContent = utils::CommonUtils::getLinkContent(&m_memoryCtx, templateLinkAddr);
```

Comment:
we can get segfolt the system by this method. Can we fix this with a separate commit?
Pls add smth like [this](https://github.com/SemSystemsInternal/patient-care/commit/0b5932e9c5e98606efbbb72ea6b13873ce67ff00)  in case of nulptr return of ms_context->GetLinkContent(scLink) in it.

Reply:
Also what is linkContent? template idtf? maybe it makes sense to rename this variable to make it easier to understand. What do you think?